src/System.Runtime.Analyzers/CSharp/CSharpDoNotRaiseReservedExceptionTypes.cs

```diff
@@ -4,16 +4,18 @@
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
-using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Diagnostics;  
 
 namespace System.Runtime.Analyzers
-{                          
+{
     /// <summary>
     /// CA2201: Do not raise reserved exception types
     /// </summary>
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
-    public sealed class CSharpDoNotRaiseReservedExceptionTypesAnalyzer : DoNotRaiseReservedExceptionTypesAnalyzer
+    public sealed class CSharpDoNotRaiseReservedExceptionTypesAnalyzer : DoNotRaiseReservedExceptionTypesAnalyzer<SyntaxKind, ObjectCreationExpressionSyntax>
     {
+        public override SyntaxKind ObjectCreationExpressionKind => SyntaxKind.ObjectCreationExpression;
```

Comment:
Interesting - this rule fires anytime one of these types is even instantiated. I would have expected a check for the throw statement. Is the idea that practically you would only instantiate exceptions to throw them?